vars/piperStageWrapper.groovy

```diff
@@ -0,0 +1,108 @@
+    }
+}
+
+private void executeStage(script, originalStage, stageName, config, utils) {
+
+    boolean projectExtensions
+    boolean globalExtensions
+    def startTime = System.currentTimeMillis()
+
+    try {
+        //Add general stage stashes to config.stashContent
+        config.stashContent += script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.unstash ?: []
+
+        utils.unstashAll(config.stashContent)
+
+        /* Defining the sources where to look for a project extension and a repository extension.
+        * Files need to be named like the executed stage to be recognized.
+        */
+        def projectInterceptorFile = "${config.projectExtensionsDirectory}${stageName}.groovy"
+        def globalInterceptorFile = "${config.globalExtensionsDirectory}${stageName}.groovy"
+        projectExtensions = fileExists(projectInterceptorFile)
+        globalExtensions = fileExists(globalInterceptorFile)
+        // Pre-defining the real originalStage in body variable, might be overwritten later if extensions exist
+        def body = originalStage
+
+        // First, check if a global extension exists via a dedicated repository
+        if (globalExtensions) {
+            Script globalInterceptorScript = load(globalInterceptorFile)
+            echo "[${STEP_NAME}] Found global interceptor '${globalInterceptorFile}' for ${stageName}."
+            // If we call the global interceptor, we will pass on originalStage as parameter
+            body = {
+                globalInterceptorScript(body, stageName, config)
+            }
+        }
+
+        // Second, check if a project extension (within the same repository) exists
+        if (projectExtensions) {
+            Script projectInterceptorScript = load(projectInterceptorFile)
+            echo "[${STEP_NAME}] Running project interceptor '${projectInterceptorFile}' for ${stageName}."
+            // If we call the project interceptor, we will pass on body as parameter which contains either originalStage or the repository interceptor
+            projectInterceptorScript(body, stageName, config)
+        } else {
+            //TODO: assign projectInterceptorScript to body as done for globalInterceptorScript, currently test framework does not seem to support this case. Further investigations needed.
+            body()
+        }
+
+    } finally {
+        echo "Current build result in stage $stageName is ${script.currentBuild.currentResult}."
+        //Perform stashing of selected files in workspace
+        utils.stashList(script, script.commonPipelineEnvironment.configuration.stageStashes?.get(stageName)?.stashes ?: [])
+        deleteDir()
+
+        def duration = System.currentTimeMillis() - startTime
+        utils.pushToSWA([eventType: 'library-os-stage', stageName: stageName, custom1: "${script.currentBuild.currentResult}", custom2: "${startTime}", custom3: "${duration}", custom4: "${projectExtensions}", custom5: "${globalExtensions}"], config)
```

Comment:
custom1, custom2, ... are currently not read in the step pushToSWA, right?
I would propose removing the analytics from here and add that together with the analytics extension pull request...

Reply:
good catch.
I mixed up the parameters. I fixed it. Reporting should now take place correctly.

Reply:
@daniel-kurzynski can we finish it with that?